### PR TITLE
Google Play release promotion

### PR DESCRIPTION
**Pull request by priitlatt**
In Google Play is is possible to promote a release from one release track to another. For example, one might initially publish the release binary (`aab` or `apk`) to be internally tested. Once is tested, then the same version could be promoted to either some broader test track like or directly to production.

This PR introduces new action `google-play tracks promote-release` to promote release from one track to another.

**Example usage:**
```shell
$ google-play tracks promote-release --package-name "io.codemagic.capybara" --source-track "internal" --target-track "alpha" --release-status "draft"
Promote release 1.0.2 (437) from track "internal" to track "alpha"
Successfully completed release promotion to track alpha
Track: alpha
Releases: [
        Status: draft
        Name: 1.0.2
        Version codes: [
                437
        ]
]
```

**New actions**
- `google-play tracks promote-release`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+Version 0.44.0
+-------------
+
+Additions and changes from [pull request #345](https://github.com/codemagic-ci-cd/cli-tools/pull/345).
+
+**Features**
+- Add new action `google-play tracks promote-release` to promote a release from one Google Play release track to another.
```
Comment (helinanever) [inline CHANGELOG.md:7]:
```suggestion
- Add a new action `google-play tracks promote-release` to promote a release from one Google Play release track to another.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+
+**Development**
+- Define new common argument type `bounded_number` for CLI usage that can be used to load floats and integers from CLI inputs within specified ranges.
```
Comment (helinanever) [inline CHANGELOG.md:10]:
```suggestion
- Define a new common argument type `bounded_number` for CLI usage that can be used to load floats and integers from CLI inputs within specified ranges.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+- Add new client method `update_track` to update release track in Google Play API client `codemagic.google_play.api_client.GooglePlayDeveloperAPIClient`.
```
Comment (helinanever) [inline CHANGELOG.md:11]:
```suggestion
- Add a new client method `update_track` to update release track in Google Play API client `codemagic.google_play.api_client.GooglePlayDeveloperAPIClient`.
```

File: docs/google-play/tracks/promote-release.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+Package name of the app in Google Play Console. For example `com.example.app`
+##### `--source-track=SOURCE_TRACK_NAME`
+
+
+Name of the track from where releases are promoted from. For example `internal`
+##### `--target-track=TARGET_TRACK_NAME`
+
+
+Name of the track to which releases are promoted to. For example `alpha`
```
Comment (helinanever) [inline docs/google-play/tracks/promote-release.md:33]:
```suggestion
Name of the track to which releases are promoted. For example `alpha`
```

File: docs/google-play/tracks/promote-release.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+### Optional arguments for action `promote-release`
+
+##### `--release-status=statusUnspecified | draft | inProgress | halted | completed`
+
+
+Promoted release status in the target track. Default:&nbsp;`completed`
```
Comment (helinanever) [inline docs/google-play/tracks/promote-release.md:39]:
```suggestion
Status of the promoted release in the target track. Default:&nbsp;`completed`
```

File: docs/google-play/tracks/promote-release.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+##### `--user-fraction=PROMOTED_USER_FRACTION`
+
+
+Fraction of users who are eligible for a staged promoted release in the target track. Number from interval `0 < fraction < 1`. Can only be set when status is `inProgress` or `halted`
+##### `--version-code-filter=PROMOTE_VERSION_CODE`
+
+
+Promote only release from source track that contains specified version code
```
Comment (helinanever) [inline docs/google-play/tracks/promote-release.md:47]:
```suggestion
Promote only a source track release that contains the specified version code
```

File: docs/google-play/tracks/promote-release.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+##### `--release-status-filter=statusUnspecified | draft | inProgress | halted | completed`
+
+
+Promote only release from source track with specified status
```
Comment (helinanever) [inline docs/google-play/tracks/promote-release.md:51]:
```suggestion
Promote only a source track release with the specified status
```

File: docs/google-play/tracks/promote-release.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+##### `--json, -j`
+
+
+Whether to show the request response in JSON format
+### Optional arguments for command `google-play`
+
+##### `--credentials=GCLOUD_SERVICE_ACCOUNT_CREDENTIALS`
+
+
+Gcloud service account credentials with `JSON` key type to access Google Play Developer API. If not given, the value will be checked from the environment variable `GCLOUD_SERVICE_ACCOUNT_CREDENTIALS`. Alternatively to entering CREDENTIALS in plaintext, it may also be specified using the `@env:` prefix followed by an environment variable name, or the `@file:` prefix followed by a path to the file containing the value. Example: `@env:<variable>` uses the value in the environment variable named `<variable>`, and `@file:<file_path>` uses the value from the file at `<file_path>`.
```
Comment (helinanever) [inline docs/google-play/tracks/promote-release.md:61]:
```suggestion
Gcloud service account credentials with the `JSON` key type to access Google Play Developer API. If not given, the value will be checked from the environment variable `GCLOUD_SERVICE_ACCOUNT_CREDENTIALS`. Alternatively to entering CREDENTIALS in plaintext, credentials may also be specified using the `@env:` prefix followed by an environment variable name, or the `@file:` prefix followed by a path to the file containing the value. Example: `@env:<variable>` uses the value in the environment variable named `<variable>`, and `@file:<file_path>` uses the value from the file at `<file_path>`.
```

File: docs/google-play/tracks/promote-release.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+### Common options
+
+##### `-h, --help`
+
+
+show this help message and exit
```
Comment (helinanever) [inline docs/google-play/tracks/promote-release.md:67]:
```suggestion
Show this help message and exit
```

**Review by helinanever — CHANGES_REQUESTED** [review]
Some nits

File: docs/google-play/tracks/promote-release.md (hunk 8 above)
Comment (priitlatt) [reply docs/google-play/tracks/promote-release.md]: Suggestion `it` -> `credentials` will not be applied. Everything related to loading values from environment variables is automatically generated and is used for arguments across all tools.

File: docs/google-play/tracks/promote-release.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+### Required arguments for action `promote-release`
+
+##### `--package-name, -p=PACKAGE_NAME`
+
+
+Package name of the app in Google Play Console. For example `com.example.app`
+##### `--source-track=SOURCE_TRACK_NAME`
+
+
+Name of the track from where releases are promoted from. For example `internal`
```
Comment (helinanever) [inline docs/google-play/tracks/promote-release.md:29]: That's double "from"
```suggestion
Name of the track from where releases are promoted. For example `internal`
```

File: src/codemagic/tools/google_play/arguments.py [hunk 11]
```diff
@@ -26,15 +28,72 @@ class TracksArgument(cli.Argument):
...
+        argparse_kwargs={"required": True},
+    )
+
+
+class PromoteArgument(cli.Argument):
+    SOURCE_TRACK_NAME = cli.ArgumentProperties(
+        key="source_track_name",
+        flags=("--source-track",),
+        description=(
+            f"Name of the track from where releases are promoted from. For example `{Colors.WHITE('internal')}`"
```
Comment (helinanever) [inline src/codemagic/tools/google_play/arguments.py:49]: same here
```suggestion
            f"Name of the track from where releases are promoted. For example `{Colors.WHITE('internal')}`"
```

**Review by helinanever — APPROVED** [review]
One last nit, but overall LGTM

File: tests/cli/argument/common_argument_types/test_bounded_number.py [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+        ("11", True),
+        ("20", True),
+        ("-1", False),
+        ("1", False),
+        ("10", False),
+        ("11", False),
+        ("20", False),
+    ),
+)
+def test_out_of_bounds(out_of_bounds_number_string, inclusive_comparison):
```
Comment (artemii-yanushevskyi) [inline tests/cli/argument/common_argument_types/test_bounded_number.py:69]: let's maybe split into two tests?
`test_out_of_inclusive_bounds` and `test_out_of_exclusive_bounds`

**Review by artemii-yanushevskyi — APPROVED** [review]

File: tests/cli/argument/common_argument_types/test_bounded_number.py (hunk 12 above)
Comment (priitlatt) [reply tests/cli/argument/common_argument_types/test_bounded_number.py]: Made the split in [2d62d6e](https://github.com/codemagic-ci-cd/cli-tools/pull/345/commits/2d62d6e218f5018ac502bcb326420771e86b260b).

**Pull request merged**